portal-frontend/src/app/shared/owner-dialogs/owner-dialog/owner-dialog.component.ts

```diff
@@ -98,26 +98,36 @@ export class OwnerDialogComponent {
   }
 
   async onCreate() {
-    if (!this.data.submissionUuid) {
-      console.error('ApplicationOwnerDialogComponent misconfigured, needs fileId for create');
-      return;
+    if (this.form.valid) {
+      if (!this.data.submissionUuid) {
+        console.error('ApplicationOwnerDialogComponent misconfigured, needs fileId for create');
+        return;
+      }
+  
+      const documentUuid = await this.uploadPendingFile(this.pendingFile);
+      const createDto: ApplicationOwnerCreateDto & NoticeOfIntentOwnerCreateDto = {
+        organizationName: this.organizationName.getRawValue() || undefined,
+        firstName: this.firstName.getRawValue() || undefined,
+        lastName: this.lastName.getRawValue() || undefined,
+        corporateSummaryUuid: documentUuid?.uuid,
+        email: this.email.getRawValue()!,
+        phoneNumber: this.phoneNumber.getRawValue()!,
+        typeCode: this.type.getRawValue()!,
+        applicationSubmissionUuid: this.data.submissionUuid,
+        noticeOfIntentSubmissionUuid: this.data.submissionUuid,
+      };
+  
+      const res = await this.data.ownerService.create(createDto);
+      this.dialogRef.close(res);
+    } else {
+      this.markFormGroupTouched(this.form)
```

Comment:
suggestion: You can use: this.form.markAllAsTouched() and you won't need the helper function.

Reply:
ohh nice, i made the changes
also added it on save
if user is trying to edit